server/app/views/admin/programs/ProgramTranslationView.java

```diff
@@ -102,55 +101,65 @@ private ImmutableList<DomContent> formFields(
                         div()
                             .with(
                                 FieldWithLabel.input()
-                                    .setFieldName("displayName")
+                                    .setFieldName(ProgramTranslationForm.DISPLAY_NAME_FORM_NAME)
                                     .setLabelText("Program name")
-                                    .setValue(translationForm.getDisplayName())
+                                    .setValue(updateData.localizedDisplayName())
                                     .getInputTag(),
                                 defaultLocaleTextHint(program.localizedName())),
                         div()
                             .with(
                                 FieldWithLabel.input()
-                                    .setFieldName("displayDescription")
+                                    .setFieldName(
+                                        ProgramTranslationForm.DISPLAY_DESCRIPTION_FORM_NAME)
                                     .setLabelText("Program description")
-                                    .setValue(translationForm.getDisplayDescription())
+                                    .setValue(updateData.localizedDisplayDescription())
                                     .getInputTag(),
                                 defaultLocaleTextHint(program.localizedDescription())))));
     if (statusTrackingEnabled.get()) {
       String programStatusesLink =
           controllers.admin.routes.AdminProgramStatusesController.index(program.id()).url();
 
-      for (StatusDefinitions.Status s : program.statusDefinitions().getStatuses()) {
+      Preconditions.checkState(
```

Comment:
If this fails it's an indication of a coding error right and not being out of sync with the FE?  We're reading the program and making the form/updates all in the same server request correct?

Reply:
Exactly. I made this a precondition check for that reason. If you'd prefer, we can throw an OutOfDateStatusesException and update the controller to handle that (though it should be unexpected). For now, I've left as-is, but I'm happy to switch to that pattern (or something else) if you'd prefer.